ivy/functional/ivy/experimental/set.py

```diff
@@ -0,0 +1,112 @@
+# global
+from typing import Union, Tuple
+
+# local
+import ivy
+from ivy.func_wrapper import (
+    handle_array_function,
+    to_native_arrays_and_back,
+    handle_nestable,
+    handle_array_like_without_promotion,
+)
+from ivy.utils.exceptions import handle_exceptions
+
+
+@handle_exceptions
+@handle_nestable
+@handle_array_like_without_promotion
+@to_native_arrays_and_back
+@handle_array_function
+def intersection(
+    x1: Union[ivy.Array, ivy.NativeArray],
+    x2: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    assume_unique: bool = False,
+    return_indices: bool = False,
+) -> Tuple[
+    Union[ivy.Array, ivy.NativeArray],
+    Union[ivy.Array, ivy.NativeArray],
+    Union[ivy.Array, ivy.NativeArray],
```

Comment:
we should be returning ivy arrays only
```suggestion
    ivy.Array,
    ivy.Array,
    ivy.Array,
```